packages/block-editor/src/components/provider/use-block-sync.js

```diff
@@ -120,17 +123,7 @@ export default function useBlockSync( {
 		let isPersistent = isLastBlockChangePersistent();
 		let previousAreBlocksDifferent = false;
 
-		onInputRef.current = onInput;
-		onChangeRef.current = onChange;
 		const unsubscribe = registry.subscribe( () => {
-			// Sometimes, subscriptions might trigger with stale callbacks
-			// before they are cleaned up. Avoid running them.
-			if (
-				onInputRef.current !== onInput ||
```

Comment:
Shouldn't we keep this? I think the `subscription` might be called more frequently than setting the refs.

Reply:
But if you ensure that the ref always contains the last version, that's not needed.

Reply:
ah, yeah, I see :)